src/app/messages-page/messages-inbox/messages-inbox.component.html

```diff
@@ -1,18 +1,46 @@
 <!-- Top Bar -->
-<div
-  *ngIf="!isMobile"
-  class="w-100 global__top-bar__height d-flex align-items-center pl-15px fs-18px font-weight-bold border-bottom border-color-grey"
->
+<div *ngIf="!isMobile"
+     class="w-100 global__top-bar__height d-flex align-items-center pl-15px fs-18px font-weight-bold border-bottom border-color-grey">
+  <!-- Clear Inbox -->
+  <div class="d-flex justify-content-end align-items-right pr-10px">
+    <div *ngIf="this.globalVars.messageResponse != null &&
+    this.globalVars.messageResponse.NumberOfUnreadThreads > 0 ; then thenEnvelopeBlock else elseEnvelopeBlock"></div>
```

Comment:
don't use `this` in your html files.

Reply:
you can simplify the conditional to be: `globalVars?.messageResponse?.NumberOfThreads`. Both null and 0 are falsy.